zonec.c

```diff
@@ -2190,183 +150,256 @@ check_dname(zone_type* zone)
 	}
 }
 
-/*
- * Reads the specified zone into the memory
- * nsd_options can be NULL if no config file is passed.
- */
-unsigned int
-zonec_read(const char* name, const char* zonefile, zone_type* zone)
+static int
+has_soa(domain_type* domain)
 {
-	const dname_type *dname;
+	rrset_type* p = NULL;
+	if(!domain) return 0;
+	for(p = domain->rrsets; p; p = p->next)
+		if(rrset_rrtype(p) == TYPE_SOA)
+			return 1;
+	return 0;
+}
 
-	totalrrs = 0;
-	startzonec = time(NULL);
-	parser->errors = 0;
+struct zonec_state {
+	struct namedb *database;
+	struct domain_table *domains;
+	struct zone *zone;
+	struct domain *domain;
+	size_t errors;
+	size_t records;
+};
+
+int32_t zonec_accept(
+	zone_parser_t *parser,
+	const zone_name_t *owner,
+	uint16_t type,
+	uint16_t class,
+	uint32_t ttl,
+	uint16_t rdlength,
+	const uint8_t *rdata,
+	void *user_data)
+{
+	struct zonec_state *state = (struct zonec_state *)user_data;
+
+	assert(state);
+
+	struct rr *rr;
+	struct rrset *rrset;
+	const struct dname *dname;
+	struct domain *domain;
+	struct buffer buffer;
+	int priority = ZONE_ERROR << (parser->options.non_strict == true);
```

Comment:
So this compares the boolean with a 'true' value, uses the numerical value of the comparison, then uses that shifts define the ZONE_ERROR and ZONE_WARNING priorities in zone.h. It could also be with '?', or is that to stop alternate code paths, for optimization? In which case that could be annotated; that non_strict has to have 'true', not unzero, and the shifts are used in this manner. Maybe use '1' instead of 'true'?

Reply:
Yeah, it's probably better to change this for readability. Will fix.